### PR TITLE
lib/game: Distributed universe

### PR DESCRIPTION
**Pull request by deadprogram**
This PR adds a new `DistributedUniverse` type, intended to be used for processing where the universes are connected via some distributed data system such as a key/value store.

### CODE REVIEW

File: lib/game/distributed.go [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+		ID:       id,
+		TopID:    NullID,
+		BottomID: NullID,
+		LeftID:   NullID,
+		RightID:  NullID,
+		Universe: NewUniverse(height, width),
+	}
+
+	d.Rules = d.rules
+	d.GetNeighbor = d.getEmptyUniverse
```
Comment (acifani) [inline lib/game/distributed.go:44]: What if we require this function to be passed? Does it make sense to have a distributed universe without being able to get neighbors?

**Review by acifani — APPROVED** [review]
LGTM!

File: lib/game/distributed.go (hunk 1 above)
Comment (deadprogram) [reply lib/game/distributed.go:44]: The default makes the universe act like it has no neighbors. That way you do not have to hookup that function just to load or save a universe.

**Comment by deadprogram** [comment]
Thank you for quick review @acifani it is appreciated!

**Pull request merged**